front/src/applications/operationalStudies/components/ManageTrainSchedule/SpeedLimitByTagSelector.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { ComponentType, useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import { updateSpeedLimitByTag } from 'reducers/osrdconf';
+import { setFailure } from 'reducers/main';
+import icon from 'assets/pictures/components/speedometer.svg';
+import SelectImprovedSNCF from 'common/BootstrapSNCF/SelectImprovedSNCF';
+import { getInfraID, getSpeedLimitByTag } from 'reducers/osrdconf/selectors';
+import { Dispatch } from 'redux';
+import { noop } from 'lodash';
+import { osrdMiddlewareApi } from 'common/api/osrdMiddlewareApi';
+import { FetchBaseQueryError } from '@reduxjs/toolkit/query/react';
+
+type SpeedLimitByTagSelectorProps = {
+  condensed?: boolean;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  speedLimitByTag?: string;
+  dispatch?: Dispatch;
+  speedLimitsByTagsFromApi?: string[];
+};
+
+function withOSRDData<T>(Component: ComponentType<T>) {
+  return (hocProps: T) => {
+    const { t } = useTranslation(['operationalStudies/manageTrainSchedule']);
+    const dispatch = useDispatch();
+    const infraID = useSelector(getInfraID);
+    const speedLimitByTag = useSelector(getSpeedLimitByTag);
+    const { data, error } = osrdMiddlewareApi.useGetInfraByIdSpeedLimitTagsQuery(
```

Comment:
```suggestion
    const { data, error } = osrdEditoastApi.useGetInfraByIdSpeedLimitTagsQuery(
```
Can you use Editoast API here ?

Reply:
Yes done